.github/workflows/test.yml

```diff
@@ -21,6 +21,9 @@ jobs:
         signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
     - run: nix-build -A checks.$(nix-instantiate --eval -E '(builtins.currentSystem)')
+    - run: |
+      NIX_BUILD=$(nix-build -A defaultPackage.$(nix-instantiate --eval -E '(builtins.currentSystem)'))/bin/nix-build
+      $NIX_BUILD --option experimental-features 'flakes' --impure --expr 'builtins.getFlake (builtins.toPath ./.).lib.testAgainst.x86_64-linux (builtins.getFlake (builtins.toPath ./.)).defaultPackage.x86_64-linux'
```

Comment:
Why is `--impure` needed?

Reply:
Both flake inputs (`./.` and `github:nixos/nix`) are mutable refs, so they aren’t allowed in pure mode